packages/core/bootstrap/src/lib/external-adapter/validator.ts

```diff
@@ -18,121 +21,84 @@ import { Requester } from './requester'
+        const inputConfig = this.inputConfigs[key]
+
+        if (Array.isArray(inputConfig)) {
+          const usedKey = this.getUsedKey(key, inputConfig)
+          if (!usedKey) this.throwInvalid(`None of aliases used for required key ${key}`)()
+          this.validateRequiredParam(usedKey as string, options)
+        } else if (typeof inputConfig === 'boolean') {
+          inputConfig
+            ? this.validateRequiredParam(key, options)
+            : this.validateOptionalParam(key, options)
+        } else if (Object.keys(inputConfig).length > 0) {
+          this.validateObjectParam(key)
         }
       }
-    } catch (error) {
-      this.parseError(
-        error,
-        {
-          input: this.input,
-          options: this.options,
-          customParams: this.customParams,
-        },
-        shouldLogError,
-      )
-    }
-  }
-
-  validateOverrides(shouldLogError: boolean) {
-    try {
-      if (!this.input.data?.overrides) {
-        this.validated.overrides = this.formatOverride(presetSymbols)
-        return
-      }
-      this.validated.overrides = this.formatOverride(
-        merge({ ...presetSymbols }, this.input.data.overrides),
-      )
     } catch (e) {
-      this.parseError(
-        e,
-        {
-          input: this.input,
-          options: this.options,
-          customParams: this.customParams,
-        },
-        shouldLogError,
-      )
+      this.parseError(e)
     }
   }
 
-  validateTokenOverrides(shouldLogError: boolean) {
+  validateOverrides(path: 'overrides' | 'tokenOverrides', preset: Record<string, any>): void {
     try {
-      if (!this.input.data?.tokenOverrides) {
-        this.validated.tokenOverrides = this.formatTokenOverrides(presetTokens)
+      if (!this.input.data?.[path]) {
+        this.validated[path] = this.formatOverride(preset)
         return
       }
-      this.validated.tokenOverrides = this.formatTokenOverrides(
-        merge({ ...presetTokens }, this.input.data.tokenOverrides),
-      )
+      this.validated[path] = this.formatOverride(merge({ ...preset }, this.input.data[path]))
     } catch (e) {
-      this.parseError(
-        e,
-        {
-          input: this.input,
-          options: this.options,
-          customParams: this.customParams,
-        },
-        shouldLogError,
-      )
+      this.parseError(e)
     }
   }
 
-  validateIncludeOverrides(shouldLogError: boolean) {
+  validateIncludeOverrides(): void {
     try {
       this.validated.includes = this.formatIncludeOverrides([
         ...(this.input.data?.includes || []),
         ...presetIncludes,
       ])
     } catch (e) {
-      this.parseError(
-        e,
-        {
-          input: this.input,
-          options: this.options,
-          customParams: this.customParams,
-        },
-        shouldLogError,
-      )
+      this.parseError(e)
     }
   }
 
-  parseError(error: any, context: any, shouldLogError: boolean) {
+  parseError(error: any): void {
```

Comment:
Does `error` not have a type?

Reply:
Looks like I should be using the `Error` type